src/components/PipelineRun/PipelineRunListView/PipelineRunsListView.tsx

```diff
@@ -107,22 +115,51 @@ const PipelineRunsListView: React.FC<React.PropsWithChildren<PipelineRunsListVie
       }
       return acc;
     }, {});
-  }, [pipelineRuns]);
+  }, [pipelineRuns, customFilter]);
+
+  const typeFilters = React.useMemo(
+    () => (typeFiltersParam ? typeFiltersParam.split(',') : []),
+    [typeFiltersParam],
+  );
+
+  const setTypeFilters = (filters: string[]) => setTypeFiltersParam(filters.join(','));
+
+  const typeFilterObj = React.useMemo(() => {
+    return pipelineRuns.reduce((acc, plr) => {
+      if (customFilter && !customFilter(plr)) {
+        return acc;
+      }
```

Comment:
This same code is being used in Other component, Create reusable utilities or component for the toolbar